### ISSUE TITLE
Integration with AbstractGPs.jl

### ISSUE THREAD

**Issue by Crown421** [SebastianCallh/IterGP.jl#3]
Hey, I am currently looking into using the IterGP algorithm and was told about this package that already contains a julia implementation. 

Would you be open to extending this package to build on top of [AbstractGPs.jl](https://juliagaussianprocesses.github.io/AbstractGPs.jl/dev/)?

Happy to do a PR, but not sure if this is a personal project/ proof of concept.

**Comment by SebastianCallh** [comment]
Hi! Thanks for the interest in this. I'm happy to properly package it up and build on AbstractGPs.jl.
 
In the interest of transparency, I want to let you know that there is currently a bug I have not been able to iron out when using the preconditioned conjugated gradient descent strategy which causes the posterior covariance to be much too large. I will look into it again, and if you feel you have the expertise I would welcome a second pair of eyes.

**Comment by SebastianCallh** [comment]
Hi @Crown421 . So I refactored the code base to use the basic AbstractGPs.jl interface (see README examples), but there are unfortunately still some rough edges. You are more than welcome to open PR to make the implementation work for your use case.
 
- When using a Preconditioned CG (in accordance to the Algorithm 1 and Table 1 in the paper), the posterior variance is wrong. Regular CG is fine so there is something going in with the preconditioner specifically.

- Trace estimation is not implemented which is needed for logpdf so currently hyperparameter tuning won't work. I added some sloppy stubs [here](https://github.com/SebastianCallh/IterGP.jl/blob/main/src/marginal_likelihood.jl) but I do not have more time to work on it right now.
